lib/include/jxl/decode.h

```diff
@@ -371,27 +372,28 @@ JXL_EXPORT size_t JxlDecoderSizeHintBasicInfo(const JxlDecoder* dec);
 JXL_EXPORT JxlDecoderStatus JxlDecoderSubscribeEvents(JxlDecoder* dec,
                                                       int events_wanted);
 
-/** Enables or disables preserving of original orientation. Some images are
- * encoded with an orientation tag indicating the image is rotated and/or
- * mirrored (here called the original orientation).
- *
- * *) If keep_orientation is JXL_FALSE (the default): the decoder will perform
- * work to undo the transformation. This ensures the decoded pixels will not
- * be rotated or mirrored. The decoder will always set the orientation field
- * of the JxlBasicInfo to JXL_ORIENT_IDENTITY to match the returned pixel data.
- * The decoder may also swap xsize and ysize in the JxlBasicInfo compared to the
- * values inside of the codestream, to correctly match the decoded pixel data,
- * e.g. when a 90 degree rotation was performed.
- *
- * *) If this option is JXL_TRUE: then the image is returned as-is, which may be
- * rotated or mirrored, and the user must check the orientation field in
- * JxlBasicInfo after decoding to correctly interpret the decoded pixel data.
+/** Enables or disables preserving of as-in-bitstream pixeldata
+ * orientation. Some images are encoded with an Orientation tag
+ * indicating that the decoder must perform a rotation and/or
+ * mirroring to the encoded image data.
+ *
+ * *) If keep_data_orientation is JXL_FALSE (the default): the decoder
+ * will apply the transformation from the orientation setting, hence
+ * rendering the image according to its specified intent. When
+ * producing a JxlBasicInfo, the decoder will always set the
+ * orientation field to JXL_ORIENT_IDENTITY (matching the returned
+ * pixel data) and also align xsize and ysize so that they correspond
+ * to the width and the height of the returned pixel data.
+ *
+ * *) If keep_data_orientation is JXL_TRUE: the decoder will skip
+ * applying the transformation from the orientation setting, returning
+ * the image in the as-in-bitstream pixeldata orientation.
  * This may be faster to decode since the decoder doesn't have to apply the
  * transformation, but can cause wrong display of the image if the orientation
  * tag is not correctly taken into account by the user.
  *
- * By default, this option is disabled, and the decoder automatically corrects
- * the orientation.
+ * By default, this option is disabled, and the decoder respects the
+ * orientation setting provided by the image.
```

Comment:
It wouldn't hurt to keep the old wording too, for maximum clarity, e.g.:
```suggestion
 * orientation setting provided by the image, i.e., it automatically orients the image correctly.
```

Reply:
I think the term "correct" should not be used in technical writing, since it basically gets used as an abbreviation for "you know what is meant". But if we both know what is meant (and that claim might actually be more contentious than it might look like): why not simply say what is meant here?

How about: "By default, this option is disabled, and after initial decoding, the pixel data gets re-oriented according to the image's Orientation setting".

Reply:
Yes, that looks good too. I'd drop the "after initial decoding" because it's an implementation detail whether we first decode pixels and then orient them, or do the orienting while decoding — what matters is what gets returned to the application.

So maybe something like this:

"By default, this option is disabled, and the returned pixel data is re-oriented according to the image's Orientation setting".

Reply:
Very good. Going with this last proposal.